---
title: "R Notebook"
output: html_notebook
---


#Get TPM read counts to filter out lowly expressed genes 

#Get 451Lu Read counts
```{r}
library(ngscmdr)
#import hit count files from RNA-seq SCC
x451Lu_1_count <- read.delim("/Users/rjfisher/Downloads/H1.counts.txt")
x451Lu_2_count <- read.delim("/Users/rjfisher/Downloads/H2.counts.txt")

#use package function to convert subread featurecounts output to TPM
x451Lu_count_1 <- calc_tpm_from_featurecounts(x451Lu_1_count)
x451Lu_count_2 <- calc_tpm_from_featurecounts(x451Lu_2_count)

#Merge replicate TPM count data
x451Lu_count <- inner_join(x451Lu_count_1, x451Lu_count_2, by = "Geneid")

#Average TPM count data and take Log2 of avg
x451Lu_count <- x451Lu_count %>%
  mutate(Avg_TPM = rowMeans(.[c("tpm_H1", "tpm_H2")], na.rm = TRUE),
         Log2_Avg_TPM = log2(Avg_TPM)) %>%
  select(Geneid, H1, H2, Avg_TPM, Log2_Avg_TPM)

#Only keep genes with TPM>=3
x451Lu_count <- x451Lu_count %>%
  filter(Log2_Avg_TPM >= 3)

#Convert Ensembl ID's in count file to gene names
library('biomaRt')
mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
genes <- x451Lu_count$Geneid
x451Lu_count<-x451Lu_count[,-4]
G_list <- getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id","hgnc_symbol"),values=genes,mart= mart)
x451Lu_count<-merge(x451Lu_count,G_list,by.x="Geneid",by.y="ensembl_gene_id")
x451Lu_count$geneSymbol <- x451Lu_count$hgnc_symbol

#Read in x451Lu rmats results and filter based on FDR < 0.05 and deltaPSI> |0.1|
x451Lu_rmats_filtered <- read.delim("/Users/rjfisher/Downloads/SE.MATS.JCEC (31).txt")
x451Lu_rmats_filtered <- x451Lu_rmats_filtered %>%
  filter(FDR < 0.05, IncLevelDifference >= 0.1 | IncLevelDifference <= -0.1)

#Merge rmats file with x451Lu count file to obtain significant splicing events based on TPM, FDR, and deltaPSI 
x451Lu_rmats_filtered <- inner_join(x451Lu_rmats_filtered, x451Lu_count, by = "geneSymbol")


x451Lu_rmats_filtered$IJC_SAMPLE_2 <- as.character(x451Lu_rmats_filtered$IJC_SAMPLE_2)  # Convert to character

# Split the values by comma, calculate the average, and create a new column
x451Lu_rmats_filtered$Averaged_IJC_SAMPLE_2 <- sapply(strsplit(x451Lu_rmats_filtered$IJC_SAMPLE_2, ","), function(x) {
  values <- as.numeric(x)
  mean(values, na.rm = TRUE)  # Calculate the average, ignoring NA values
})

x451Lu_rmats_filtered$SJC_SAMPLE_2 <- as.character(x451Lu_rmats_filtered$SJC_SAMPLE_2)  # Convert to character

# Split the values by comma, calculate the average, and create a new column
x451Lu_rmats_filtered$Averaged_SJC_SAMPLE_2 <- sapply(strsplit(x451Lu_rmats_filtered$SJC_SAMPLE_2, ","), function(x) {
  values <- as.numeric(x)
  mean(values, na.rm = TRUE)  # Calculate the average, ignoring NA values
})

x451Lu_rmats_filtered <- x451Lu_rmats_filtered %>%
  filter(Averaged_IJC_SAMPLE_2 >= 20 | Averaged_SJC_SAMPLE_2 >= 20)


write.csv(x451Lu_rmats_filtered, file = "451Lu_rmats_filtered.csv")
```



#Get 1205Lu Read counts
```{r}
library(ngscmdr)
#import hit count files from RNA-seq SCC
x1205Lu_1_count <- read.delim("/Users/rjfisher/Downloads/G1.counts.txt")
x1205Lu_2_count <- read.delim("/Users/rjfisher/Downloads/G2.counts.txt")

#use package function to convert subread featurecounts output to TPM
x1205Lu_count_1 <- calc_tpm_from_featurecounts(x1205Lu_1_count)
x1205Lu_count_2 <- calc_tpm_from_featurecounts(x1205Lu_2_count)

#Merge replicate TPM count data
x1205Lu_count <- inner_join(x1205Lu_count_1, x1205Lu_count_2, by = "Geneid")

#Average TPM count data and take Log2 of avg
x1205Lu_count <- x1205Lu_count %>%
  mutate(Avg_TPM = rowMeans(.[c("tpm_G1", "tpm_G2")], na.rm = TRUE),
         Log2_Avg_TPM = log2(Avg_TPM)) %>%
  select(Geneid, G1, G2, Avg_TPM, Log2_Avg_TPM)

#Only keep genes with TPM>=3
x1205Lu_count <- x1205Lu_count %>%
  filter(Log2_Avg_TPM >= 3)

#Convert Ensembl ID's in count file to gene names
library('biomaRt')
mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
genes <- x1205Lu_count$Geneid
x1205Lu_count<-x1205Lu_count[,-4]
G_list <- getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id","hgnc_symbol"),values=genes,mart= mart)
x1205Lu_count<-merge(x1205Lu_count,G_list,by.x="Geneid",by.y="ensembl_gene_id")
x1205Lu_count$geneSymbol <- x1205Lu_count$hgnc_symbol

#Read in x1205Lu rmats results and filter based on FDR < 0.05 and deltaPSI> |0.1|
x1205Lu_rmats_filtered <- read.delim("/Users/rjfisher/Downloads/SE.MATS.JCEC (32).txt")
x1205Lu_rmats_filtered <- x1205Lu_rmats_filtered %>%
  filter(FDR < 0.05, IncLevelDifference >= 0.1 | IncLevelDifference <= -0.1)

#Merge rmats file with x1205Lu count file to obtain significant splicing events based on TPM, FDR, and deltaPSI 
x1205Lu_rmats_filtered <- inner_join(x1205Lu_rmats_filtered, x1205Lu_count, by = "geneSymbol")


x1205Lu_rmats_filtered$IJC_SAMPLE_2 <- as.character(x1205Lu_rmats_filtered$IJC_SAMPLE_2)  # Convert to character

# Split the values by comma, calculate the average, and create a new column
x1205Lu_rmats_filtered$Averaged_IJC_SAMPLE_2 <- sapply(strsplit(x1205Lu_rmats_filtered$IJC_SAMPLE_2, ","), function(x) {
  values <- as.numeric(x)
  mean(values, na.rm = TRUE)  # Calculate the average, ignoring NA values
})

x1205Lu_rmats_filtered$SJC_SAMPLE_2 <- as.character(x1205Lu_rmats_filtered$SJC_SAMPLE_2)  # Convert to character

# Split the values by comma, calculate the average, and create a new column
x1205Lu_rmats_filtered$Averaged_SJC_SAMPLE_2 <- sapply(strsplit(x1205Lu_rmats_filtered$SJC_SAMPLE_2, ","), function(x) {
  values <- as.numeric(x)
  mean(values, na.rm = TRUE)  # Calculate the average, ignoring NA values
})

x1205Lu_rmats_filtered <- x1205Lu_rmats_filtered %>%
  filter(Averaged_IJC_SAMPLE_2 >= 20 | Averaged_SJC_SAMPLE_2 >= 20)


write.csv(x1205Lu_rmats_filtered, file = "1205Lu_rmats_filtered.csv")
```


#Get WM793 Read counts
```{r}
library(ngscmdr)
#import hit count files from RNA-seq SCC
WM793_1_count <- read.delim("/Users/rjfisher/Downloads/I1.counts.txt")
WM793_2_count <- read.delim("/Users/rjfisher/Downloads/I2.counts.txt")

#use package function to convert subread featurecounts output to TPM
WM793_count_1 <- calc_tpm_from_featurecounts(WM793_1_count)
WM793_count_2 <- calc_tpm_from_featurecounts(WM793_2_count)

#Merge replicate TPM count data
WM793_count <- inner_join(WM793_count_1, WM793_count_2, by = "Geneid")

#Average TPM count data and take Log2 of avg
WM793_count <- WM793_count %>%
  mutate(Avg_TPM = rowMeans(.[c("tpm_I1", "tpm_I2")], na.rm = TRUE),
         Log2_Avg_TPM = log2(Avg_TPM)) %>%
  select(Geneid, I1, I2, Avg_TPM, Log2_Avg_TPM)

#Only keep genes with TPM>=3
WM793_count <- WM793_count %>%
  filter(Log2_Avg_TPM >= 3)

#Convert Ensembl ID's in count file to gene names
library('biomaRt')
mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
genes <- WM793_count$Geneid
WM793_count<-WM793_count[,-4]
G_list <- getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id","hgnc_symbol"),values=genes,mart= mart)
WM793_count<-merge(WM793_count,G_list,by.x="Geneid",by.y="ensembl_gene_id")
WM793_count$geneSymbol <- WM793_count$hgnc_symbol

#Read in WM793 rmats results and filter based on FDR < 0.05 and deltaPSI> |0.1|
WM793_rmats_filtered <- read.delim("/Users/rjfisher/Downloads/SE.MATS.JCEC (33).txt")
WM793_rmats_filtered <- WM793_rmats_filtered %>%
  filter(FDR < 0.05, IncLevelDifference >= 0.1 | IncLevelDifference <= -0.1)

#Merge rmats file with WM793 count file to obtain significant splicing events based on TPM, FDR, and deltaPSI 
WM793_rmats_filtered <- inner_join(WM793_rmats_filtered, WM793_count, by = "geneSymbol")


WM793_rmats_filtered$IJC_SAMPLE_2 <- as.character(WM793_rmats_filtered$IJC_SAMPLE_2)  # Convert to character

# Split the values by comma, calculate the average, and create a new column
WM793_rmats_filtered$Averaged_IJC_SAMPLE_2 <- sapply(strsplit(WM793_rmats_filtered$IJC_SAMPLE_2, ","), function(x) {
  values <- as.numeric(x)
  mean(values, na.rm = TRUE)  # Calculate the average, ignoring NA values
})

WM793_rmats_filtered$SJC_SAMPLE_2 <- as.character(WM793_rmats_filtered$SJC_SAMPLE_2)  # Convert to character

# Split the values by comma, calculate the average, and create a new column
WM793_rmats_filtered$Averaged_SJC_SAMPLE_2 <- sapply(strsplit(WM793_rmats_filtered$SJC_SAMPLE_2, ","), function(x) {
  values <- as.numeric(x)
  mean(values, na.rm = TRUE)  # Calculate the average, ignoring NA values
})

WM793_rmats_filtered <- WM793_rmats_filtered %>%
  filter(Averaged_IJC_SAMPLE_2 >= 20 | Averaged_SJC_SAMPLE_2 >= 20)


write.csv(WM793_rmats_filtered, file = "WM793_rmats_filtered.csv")
```



#Get SKMEL24 Read counts
```{r}
library(ngscmdr)
#import hit count files from RNA-seq SCC
SKMEL24_1_count <- read.delim("/Users/rjfisher/Downloads/K1.counts.txt")
SKMEL24_2_count <- read.delim("/Users/rjfisher/Downloads/K2.counts.txt")

#use package function to convert subread featurecounts output to TPM
SKMEL24_count_1 <- calc_tpm_from_featurecounts(SKMEL24_1_count)
SKMEL24_count_2 <- calc_tpm_from_featurecounts(SKMEL24_2_count)

#Merge replicate TPM count data
SKMEL24_count <- inner_join(SKMEL24_count_1, SKMEL24_count_2, by = "Geneid")

#Average TPM count data and take Log2 of avg
SKMEL24_count <- SKMEL24_count %>%
  mutate(Avg_TPM = rowMeans(.[c("tpm_K1", "tpm_K2")], na.rm = TRUE),
         Log2_Avg_TPM = log2(Avg_TPM)) %>%
  select(Geneid, K1, K2, Avg_TPM, Log2_Avg_TPM)

#Only keep genes with TPM>=3
SKMEL24_count <- SKMEL24_count %>%
  filter(Log2_Avg_TPM >= 3)

#Convert Ensembl ID's in count file to gene names
library('biomaRt')
mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
genes <- SKMEL24_count$Geneid
SKMEL24_count<-SKMEL24_count[,-4]
G_list <- getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id","hgnc_symbol"),values=genes,mart= mart)
SKMEL24_count<-merge(SKMEL24_count,G_list,by.x="Geneid",by.y="ensembl_gene_id")
SKMEL24_count$geneSymbol <- SKMEL24_count$hgnc_symbol

#Read in SKMEL24 rmats results and filter based on FDR < 0.05 and deltaPSI> |0.1|
SKMEL24_rmats_filtered <- read.delim("/Users/rjfisher/Downloads/SE.MATS.JCEC (34).txt")
SKMEL24_rmats_filtered <- SKMEL24_rmats_filtered %>%
  filter(FDR < 0.05, IncLevelDifference >= 0.1 | IncLevelDifference <= -0.1)

#Merge rmats file with SKMEL24 count file to obtain significant splicing events based on TPM, FDR, and deltaPSI 
SKMEL24_rmats_filtered <- inner_join(SKMEL24_rmats_filtered, SKMEL24_count, by = "geneSymbol")


SKMEL24_rmats_filtered$IJC_SAMPLE_2 <- as.character(SKMEL24_rmats_filtered$IJC_SAMPLE_2)  # Convert to character

# Split the values by comma, calculate the average, and create a new column
SKMEL24_rmats_filtered$Averaged_IJC_SAMPLE_2 <- sapply(strsplit(SKMEL24_rmats_filtered$IJC_SAMPLE_2, ","), function(x) {
  values <- as.numeric(x)
  mean(values, na.rm = TRUE)  # Calculate the average, ignoring NA values
})

SKMEL24_rmats_filtered$SJC_SAMPLE_2 <- as.character(SKMEL24_rmats_filtered$SJC_SAMPLE_2)  # Convert to character

# Split the values by comma, calculate the average, and create a new column
SKMEL24_rmats_filtered$Averaged_SJC_SAMPLE_2 <- sapply(strsplit(SKMEL24_rmats_filtered$SJC_SAMPLE_2, ","), function(x) {
  values <- as.numeric(x)
  mean(values, na.rm = TRUE)  # Calculate the average, ignoring NA values
})

SKMEL24_rmats_filtered <- SKMEL24_rmats_filtered %>%
  filter(Averaged_IJC_SAMPLE_2 >= 20 | Averaged_SJC_SAMPLE_2 >= 20)


write.csv(SKMEL24_rmats_filtered, file = "SKMEL24_rmats_filtered.csv")
```



#Get SKMEL28 Read counts
```{r}
library(ngscmdr)
#import hit count files from RNA-seq SCC
SKMEL28_1_count <- read.delim("/Users/rjfisher/Downloads/3.counts.txt")
SKMEL28_2_count <- read.delim("/Users/rjfisher/Downloads/4.counts.txt")

#use package function to convert subread featurecounts output to TPM
SKMEL28_count_1 <- calc_tpm_from_featurecounts(SKMEL28_1_count)
SKMEL28_count_2 <- calc_tpm_from_featurecounts(SKMEL28_2_count)

#Merge replicate TPM count data
SKMEL28_count <- inner_join(SKMEL28_count_1, SKMEL28_count_2, by = "Geneid")

#Average TPM count data and take Log2 of avg
SKMEL28_count <- SKMEL28_count %>%
  mutate(Avg_TPM = rowMeans(.[c("tpm_X3", "tpm_X4")], na.rm = TRUE),
         Log2_Avg_TPM = log2(Avg_TPM)) %>%
  dplyr::select(Geneid, X3, X4, Avg_TPM, Log2_Avg_TPM)

#Only keep genes with TPM>=3
SKMEL28_count <- SKMEL28_count %>%
  filter(Log2_Avg_TPM >= 3)

#Convert Ensembl ID's in count file to gene names
library('biomaRt')
mart <- useDataset("hsapiens_gene_ensembl", useMart("ensembl"))
genes <- SKMEL28_count$Geneid
SKMEL28_count<-SKMEL28_count[,-4]
G_list <- getBM(filters= "ensembl_gene_id", attributes= c("ensembl_gene_id","hgnc_symbol"),values=genes,mart= mart)
SKMEL28_count<-merge(SKMEL28_count,G_list,by.x="Geneid",by.y="ensembl_gene_id")
SKMEL28_count$geneSymbol <- SKMEL28_count$hgnc_symbol

#Read in SKMEL28 rmats results and filter based on FDR < 0.05 and deltaPSI> |0.1|
SKMEL28_rmats_filtered <- read.delim("/Users/rjfisher/Downloads/SE.MATS.JCEC-2.txt")
SKMEL28_rmats_filtered <- SKMEL28_rmats_filtered %>%
  filter(FDR < 0.05, IncLevelDifference >= 0.1 | IncLevelDifference <= -0.1)

#Merge rmats file with SKMEL28 count file to obtain significant splicing events based on TPM, FDR, and deltaPSI 
SKMEL28_rmats_filtered <- inner_join(SKMEL28_rmats_filtered, SKMEL28_count, by = "geneSymbol")


SKMEL28_rmats_filtered$IJC_SAMPLE_2 <- as.character(SKMEL28_rmats_filtered$IJC_SAMPLE_2)  # Convert to character

# Split the values by comma, calculate the average, and create a new column
SKMEL28_rmats_filtered$Averaged_IJC_SAMPLE_2 <- sapply(strsplit(SKMEL28_rmats_filtered$IJC_SAMPLE_2, ","), function(x) {
  values <- as.numeric(x)
  mean(values, na.rm = TRUE)  # Calculate the average, ignoring NA values
})

SKMEL28_rmats_filtered$SJC_SAMPLE_2 <- as.character(SKMEL28_rmats_filtered$SJC_SAMPLE_2)  # Convert to character

# Split the values by comma, calculate the average, and create a new column
SKMEL28_rmats_filtered$Averaged_SJC_SAMPLE_2 <- sapply(strsplit(SKMEL28_rmats_filtered$SJC_SAMPLE_2, ","), function(x) {
  values <- as.numeric(x)
  mean(values, na.rm = TRUE)  # Calculate the average, ignoring NA values
})

SKMEL28_rmats_filtered <- SKMEL28_rmats_filtered %>%
  filter(Averaged_IJC_SAMPLE_2 >= 20 | Averaged_SJC_SAMPLE_2 >= 20)


write.csv(SKMEL28_rmats_filtered, file = "SKMEL28_rmats_filtered.csv")
```


#SKMEL5
```{r}
# Read the FASTA file containing amino acid sequences
fasta_file <- "/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/SpliceTools/Splicetools Outputs/Neopeptides/SE_NEOPEPTIDES_Filtered7_copy.txt"
sequences <- readAAStringSet(fasta_file)

# Define a function to generate all k-mers of a given length
get_kmers <- function(sequence, k) {
  seq_len <- nchar(sequence)
  kmers <- character(0)  # Initialize an empty character vector
  
  for (i in 1:(seq_len - k + 1)) {
    kmer <- substr(sequence, i, i + k - 1)
    kmers <- c(kmers, kmer)
  }
  
  return(kmers)
}


# Define the desired k-mer lengths (8, 9, 10, and 11)
kmer_lengths <- c(8, 9, 10, 11)

# Create a list to store all k-mers for each sequence and length
all_kmers <- list()

sequences_as_character <- as.character(sequences)

# Iterate over each sequence and each k-mer length
for (sequence in sequences_as_character) {
  for (k in kmer_lengths) {
    kmers <- get_kmers(sequence, k)
    all_kmers[[paste0("k", k)]] <- c(all_kmers[[paste0("k", k)]], kmers)
  }
}

total_sequences <- sum(lengths(all_kmers))
cat("Total number of 8-11mer neopeptide sequences in SKMEL5 all_kmers:", total_sequences, "\n")
all_kmers_SKMEL5 <- all_kmers
```

#451Lu
```{r}
# Read the FASTA file containing amino acid sequences
fasta_file <- "/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/SpliceTools/Splicetools Outputs/Neopeptides/SE_NEOPEPTIDES.451Lu.filtered7.txt"
sequences <- readAAStringSet(fasta_file)

# Define a function to generate all k-mers of a given length
get_kmers <- function(sequence, k) {
  seq_len <- nchar(sequence)
  kmers <- character(0)  # Initialize an empty character vector
  
  for (i in 1:(seq_len - k + 1)) {
    kmer <- substr(sequence, i, i + k - 1)
    kmers <- c(kmers, kmer)
  }
  
  return(kmers)
}


# Define the desired k-mer lengths (8, 9, 10, and 11)
kmer_lengths <- c(8, 9, 10, 11)

# Create a list to store all k-mers for each sequence and length
all_kmers <- list()

sequences_as_character <- as.character(sequences)

# Iterate over each sequence and each k-mer length
for (sequence in sequences_as_character) {
  for (k in kmer_lengths) {
    kmers <- get_kmers(sequence, k)
    all_kmers[[paste0("k", k)]] <- c(all_kmers[[paste0("k", k)]], kmers)
  }
}

total_sequences <- sum(lengths(all_kmers))
cat("Total number of 8-11mer neopeptide sequences in 451Lu all_kmers:", total_sequences, "\n")
all_kmers_451Lu <- all_kmers

```

#1205Lu
```{r}
# Read the FASTA file containing amino acid sequences
fasta_file <- "/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/SpliceTools/Splicetools Outputs/Neopeptides/SE_NEOPEPTIDES.1205Lu.filtered7.txt"
sequences <- readAAStringSet(fasta_file)

# Define a function to generate all k-mers of a given length
get_kmers <- function(sequence, k) {
  seq_len <- nchar(sequence)
  kmers <- character(0)  # Initialize an empty character vector
  
  for (i in 1:(seq_len - k + 1)) {
    kmer <- substr(sequence, i, i + k - 1)
    kmers <- c(kmers, kmer)
  }
  
  return(kmers)
}


# Define the desired k-mer lengths (8, 9, 10, and 11)
kmer_lengths <- c(8, 9, 10, 11)

# Create a list to store all k-mers for each sequence and length
all_kmers <- list()

sequences_as_character <- as.character(sequences)

# Iterate over each sequence and each k-mer length
for (sequence in sequences_as_character) {
  for (k in kmer_lengths) {
    kmers <- get_kmers(sequence, k)
    all_kmers[[paste0("k", k)]] <- c(all_kmers[[paste0("k", k)]], kmers)
  }
}

total_sequences <- sum(lengths(all_kmers))
cat("Total number of 8-11mer neopeptide sequences in 1205Lu all_kmers:", total_sequences, "\n")
all_kmers_1205Lu <- all_kmers

```


#SKMEL24
```{r}
# Read the FASTA file containing amino acid sequences
fasta_file <- "/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/SpliceTools/Splicetools Outputs/Neopeptides/SE_NEOPEPTIDES.SKMEL24.filtered7.txt"
sequences <- readAAStringSet(fasta_file)

# Define a function to generate all k-mers of a given length
get_kmers <- function(sequence, k) {
  seq_len <- nchar(sequence)
  kmers <- character(0)  # Initialize an empty character vector
  
  for (i in 1:(seq_len - k + 1)) {
    kmer <- substr(sequence, i, i + k - 1)
    kmers <- c(kmers, kmer)
  }
  
  return(kmers)
}


# Define the desired k-mer lengths (8, 9, 10, and 11)
kmer_lengths <- c(8, 9, 10, 11)

# Create a list to store all k-mers for each sequence and length
all_kmers <- list()

sequences_as_character <- as.character(sequences)

# Iterate over each sequence and each k-mer length
for (sequence in sequences_as_character) {
  for (k in kmer_lengths) {
    kmers <- get_kmers(sequence, k)
    all_kmers[[paste0("k", k)]] <- c(all_kmers[[paste0("k", k)]], kmers)
  }
}

total_sequences <- sum(lengths(all_kmers))
cat("Total number of 8-11mer neopeptide sequences in SKMEL24 all_kmers:", total_sequences, "\n")
all_kmers_SKMEL24 <- all_kmers

```

#SKMEL28
```{r}
# Read the FASTA file containing amino acid sequences
fasta_file <- "/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/SpliceTools/Splicetools Outputs/Neopeptides/SE_NEOPEPTIDES.SKMEL28.filtered7.txt"
sequences <- readAAStringSet(fasta_file)

# Define a function to generate all k-mers of a given length
get_kmers <- function(sequence, k) {
  seq_len <- nchar(sequence)
  kmers <- character(0)  # Initialize an empty character vector
  
  for (i in 1:(seq_len - k + 1)) {
    kmer <- substr(sequence, i, i + k - 1)
    kmers <- c(kmers, kmer)
  }
  
  return(kmers)
}


# Define the desired k-mer lengths (8, 9, 10, and 11)
kmer_lengths <- c(8, 9, 10, 11)

# Create a list to store all k-mers for each sequence and length
all_kmers <- list()

sequences_as_character <- as.character(sequences)

# Iterate over each sequence and each k-mer length
for (sequence in sequences_as_character) {
  for (k in kmer_lengths) {
    kmers <- get_kmers(sequence, k)
    all_kmers[[paste0("k", k)]] <- c(all_kmers[[paste0("k", k)]], kmers)
  }
}

total_sequences <- sum(lengths(all_kmers))
cat("Total number of 8-11mer neopeptide sequences in SKMEL28 all_kmers:", total_sequences, "\n")
all_kmers_SKMEL28 <- all_kmers

```

#WM793
```{r}
# Read the FASTA file containing amino acid sequences
fasta_file <- "/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/SpliceTools/Splicetools Outputs/Neopeptides/SE_NEOPEPTIDES.WM793.filtered7.txt"
sequences <- readAAStringSet(fasta_file)

# Define a function to generate all k-mers of a given length
get_kmers <- function(sequence, k) {
  seq_len <- nchar(sequence)
  kmers <- character(0)  # Initialize an empty character vector
  
  for (i in 1:(seq_len - k + 1)) {
    kmer <- substr(sequence, i, i + k - 1)
    kmers <- c(kmers, kmer)
  }
  
  return(kmers)
}


# Define the desired k-mer lengths (8, 9, 10, and 11)
kmer_lengths <- c(8, 9, 10, 11)

# Create a list to store all k-mers for each sequence and length
all_kmers <- list()

sequences_as_character <- as.character(sequences)

# Iterate over each sequence and each k-mer length
for (sequence in sequences_as_character) {
  for (k in kmer_lengths) {
    kmers <- get_kmers(sequence, k)
    all_kmers[[paste0("k", k)]] <- c(all_kmers[[paste0("k", k)]], kmers)
  }
}

total_sequences <- sum(lengths(all_kmers))
cat("Total number of 8-11mer neopeptide sequences in WM793 all_kmers:", total_sequences, "\n")
all_kmers_WM793 <- all_kmers

```




```{r}
# Initialize an empty dataframe to store the combined gene information
combined_gene_info <- data.frame(Header = character(0), Sequence = character(0))

# Iterate over each k-mer length
for (k in kmer_lengths) {
  # Merge the current k-mer dataframe with the combined_gene_info dataframe
  combined_gene_info <- merge(combined_gene_info, all_kmers[[paste0("k", k)]], 
                              by = c("Header", "Sequence"), all = TRUE)
}

# Print the combined gene information
print(combined_gene_info)

```


```{r}
combined_kmers_WM793 <- unlist(lapply(all_kmers_WM793[c("k8", "k9", "k10", "k11")], unlist))
combined_kmers_451Lu <- unlist(lapply(all_kmers_451Lu[c("k8", "k9", "k10", "k11")], unlist))
combined_kmers_1205Lu <- unlist(lapply(all_kmers_1205Lu[c("k8", "k9", "k10", "k11")], unlist))
combined_kmers_SKMEL5 <- unlist(lapply(all_kmers_SKMEL5[c("k8", "k9", "k10", "k11")], unlist))
combined_kmers_SKMEL24 <- unlist(lapply(all_kmers_SKMEL24[c("k8", "k9", "k10", "k11")], unlist))
combined_kmers_SKMEL28 <- unlist(lapply(all_kmers_SKMEL28[c("k8", "k9", "k10", "k11")], unlist))
```

```{r}
SKMEL5_all_neopeptides <- as.data.frame(combined_kmers_SKMEL5)
SKMEL24_all_neopeptides <- as.data.frame(combined_kmers_SKMEL24)
SKMEL28_all_neopeptides <- as.data.frame(combined_kmers_SKMEL28)
WM793_all_neopeptides <- as.data.frame(combined_kmers_WM793)
all_neopeptides_1205Lu <- as.data.frame(combined_kmers_1205Lu)
all_neopeptides_451Lu <- as.data.frame(combined_kmers_451Lu)

```

```{r}
SKMEL28_all_neopeptides <- SKMEL28_all_neopeptides %>% distinct(combined_kmers_SKMEL28)
SKMEL24_all_neopeptides <- SKMEL24_all_neopeptides %>% distinct(combined_kmers_SKMEL24)
SKMEL5_all_neopeptides <- SKMEL5_all_neopeptides %>% distinct(combined_kmers_SKMEL5)
WM793_all_neopeptides <- WM793_all_neopeptides %>% distinct(combined_kmers_WM793)
all_neopeptides_1205Lu <- all_neopeptides_1205Lu %>% distinct(combined_kmers_1205Lu)
all_neopeptides_451Lu <- all_neopeptides_451Lu %>% distinct(combined_kmers_451Lu)
```

```{r}
write.csv(SKMEL5_all_neopeptides, file = "/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Peptides/SKMEL5_all_neopeptides.csv")
write.csv(SKMEL24_all_neopeptides, file = "/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Peptides/SKMEL24_all_neopeptides.csv")
write.csv(SKMEL28_all_neopeptides, file = "/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Peptides/SKMEL28_all_neopeptides.csv")
write.csv(WM793_all_neopeptides, file = "/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Peptides/WM793_all_neopeptides.csv")
write.csv(all_neopeptides_451Lu, file = "/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Peptides/all_neopeptides_451Lu.csv")
write.csv(all_neopeptides_1205Lu, file = "/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Peptides/all_neopeptides_1205Lu.csv")
```


#Find corin exclusive peptides & neopeptides

```{r}
DMSO_1 <- read.csv("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Neo_IPMS/72h_5perc/DMSO_2.csv")
DMSO_2 <- read.csv("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Neo_IPMS/72h_5perc/DMSO_2.csv")
CORIN_1 <- read.csv("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Neo_IPMS/72h_5perc/CORIN_1.csv")
CORIN_2 <- read.csv("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Neo_IPMS/72h_5perc/CORIN_2.csv")
```

```{r}
CORIN <- rbind(CORIN_1, CORIN_2)
CORIN_only_5perc <- anti_join(CORIN, DMSO_1, by = "Peptide")
CORIN_only_5perc <- anti_join(CORIN, DMSO_2, by = "Peptide")
CORIN_only_5perc <- CORIN_only_5perc %>% distinct(Peptide)
```

```{r}
write.csv(CORIN_only_5perc,file = "IPMS_CORIN_only_5perc.csv")
```

```{r}
corin_exc_peptides <- read.csv("/projectnb/epigen/Robert/melanoma_splice/SNAF/sing/fasta/IPMS_CORIN_only_5perc.csv")
corin_exc_peptides <- corin_exc_peptides$Peptide[nchar(corin_exc_peptides$Peptide) <= 11]
corin_exc_peptides <- as.data.frame(corin_exc_peptides)
corin_exc_peptides <- corin_exc_peptides %>% rename(Peptide = corin_exc_peptides)
corin_exc_peptides <- corin_exc_peptides %>% 
                distinct(Peptide, .keep_all = TRUE)
```

```{r}
write.csv(corin_exc_peptides, "corin_exc_peptides.csv")
```

```{r}
# Function to extract peptide sequences from a FASTA file
extract_sequences_to_dataframe <- function(fasta_file) {
  # Read lines from the FASTA file
  lines <- readLines(fasta_file)
  
  # Extract sequences
  sequences <- lines[!grepl("^>", lines)]  # Select lines not starting with '>'
  
  # Create a dataframe with one column for sequences
  df <- data.frame(Peptide = sequences)
  
  return(df)
}
```

```{r}
fasta_file <- "/projectnb/epigen/Robert/melanoma_splice/SNAF/sing/fasta/ucsc_proteomeFR_customVirus_8_12mer.fasta"
human_peptides <- extract_sequences_to_dataframe(fasta_file)
```

```{r}
corin_neopeptides <- anti_join(corin_exc_peptides,human_peptides, by = "Peptide")
corin_neopeptides <- corin_neopeptides %>% distinct(Peptide)
```

```{r}
write.csv(corin_neopeptides, file = "corin_neopeptides.csv")
```



#identify neopeptides from each cell line based on human proteome
```{r}
splicetools_SKMEL5 <- read.csv("/projectnb/epigen/Robert/melanoma_splice/SNAF/sing/fasta/SKMEL5_all_neopeptides.csv")
splicetools_SKMEL24 <- read.csv("/projectnb/epigen/Robert/melanoma_splice/SNAF/sing/fasta/SKMEL24_all_neopeptides.csv")
splicetools_SKMEL28 <- read.csv("/projectnb/epigen/Robert/melanoma_splice/SNAF/sing/fasta/SKMEL28_all_neopeptides.csv")
splicetools_WM793 <- read.csv("/projectnb/epigen/Robert/melanoma_splice/SNAF/sing/fasta/WM793_all_neopeptides.csv")
splicetools_1205Lu <- read.csv("/projectnb/epigen/Robert/melanoma_splice/SNAF/sing/fasta/all_neopeptides_1205Lu.csv")
splicetools_451Lu <- read.csv("/projectnb/epigen/Robert/melanoma_splice/SNAF/sing/fasta/all_neopeptides_451Lu.csv")
```

```{r}
splicetools_SKMEL5 <- splicetools_SKMEL5 %>% select(combined_kmers_SKMEL5) %>% rename(Peptide = combined_kmers_SKMEL5)
splicetools_SKMEL24 <- splicetools_SKMEL24 %>% select(combined_kmers_SKMEL24) %>% rename(Peptide = combined_kmers_SKMEL24)
splicetools_SKMEL28 <- splicetools_SKMEL28 %>% select(combined_kmers_SKMEL28) %>% rename(Peptide = combined_kmers_SKMEL28)
splicetools_WM793 <- splicetools_WM793 %>% select(combined_kmers_WM793) %>% rename(Peptide = combined_kmers_WM793)
splicetools_1205Lu <- splicetools_1205Lu %>% select(combined_kmers_1205Lu) %>% rename(Peptide = combined_kmers_1205Lu)
splicetools_451Lu <- splicetools_451Lu %>% select(combined_kmers_451Lu) %>% rename(Peptide = combined_kmers_451Lu)

```

```{r}
splicetools_neopeptides_SKMEL5 <- anti_join(splicetools_SKMEL5, human_peptides, by = "Peptide")
splicetools_neopeptides_SKMEL24 <- anti_join(splicetools_SKMEL24, human_peptides, by = "Peptide")
splicetools_neopeptides_SKMEL28 <- anti_join(splicetools_SKMEL28, human_peptides, by = "Peptide")
splicetools_neopeptides_WM793 <- anti_join(splicetools_WM793, human_peptides, by = "Peptide")
splicetools_neopeptides_451Lu <- anti_join(splicetools_451Lu, human_peptides, by = "Peptide")
splicetools_neopeptides_1205Lu <- anti_join(splicetools_1205Lu, human_peptides, by = "Peptide")
```



#Goal - combine SNAF data with splicetools data

# 1) Need to import in SNAF peptides and ensure we only are looking at neopeptides - those not present in the human proteome
```{r}
setwd("/projectnb/epigen/Robert/melanoma_splice/SNAF/sing/result")
SNAF_peptides <- read.csv("SNAF_peptides.csv")
```

#identify all neopeptides
```{r}
SNAF_neopeptides <- anti_join(SNAF_peptides, human_peptides, by = "Peptide")
```

#filter out dPSI < |0.1|
```{r}
setwd("/projectnb/epigen/Robert/melanoma_splice/SNAF/sing/result")
SNAF_PSI <- read.csv("SNAF_PSI.csv")
SNAF_PSI <- SNAF_PSI %>% rename(Gene = Region) %>% filter(abs(dPSI) > 0.1) %>% dplyr::select(Gene, dPSI)
SNAF_neopeptides_try <- inner_join(SNAF_neopeptides, SNAF_PSI, by = "Gene")
SNAF_neopeptides_try <- unique(SNAF_neopeptides_try)
```

#Cell line tables - include neopeptides found in both corin treated samples but not either DMSO
```{r}
#SKMEL5
SNAF_SKMEL5_neopeptide <- SNAF_neopeptides_try %>%
  filter(grepl('J1.bed', samples) & grepl('J2.bed', samples) & !grepl('D1.bed', samples) & !grepl('D2.bed', samples))

#SKMEL24
SNAF_SKMEL24_neopeptide <- SNAF_neopeptides_try %>%
  filter(grepl('K1.bed', samples) & grepl('K2.bed', samples) & !grepl('E1.bed', samples) & !grepl('E2.bed', samples))

#SKMEL28
SNAF_SKMEL28_neopeptide <- SNAF_neopeptides_try %>%
  filter(grepl('3.bed', samples) & grepl('4.bed', samples) & !grepl('1.bed', samples) & !grepl('2.bed', samples))

#451Lu
SNAF_451Lu_neopeptide <- SNAF_neopeptides_try %>%
  filter(grepl('H1.bed', samples) & grepl('H2.bed', samples) & !grepl('B1.bed', samples) & !grepl('B2.bed', samples))

#1205Lu
SNAF_1205Lu_neopeptide <- SNAF_neopeptides_try %>%
  filter(grepl('G1.bed', samples) & grepl('G2.bed', samples) & !grepl('A1.bed', samples) & !grepl('A2.bed', samples))

#WM793
SNAF_WM793_neopeptide <- SNAF_neopeptides_try %>%
  filter(grepl('I1.bed', samples) & grepl('I2.bed', samples) & !grepl('C1.bed', samples) & !grepl('C2.bed', samples))
```

# 2) combine SNAF and splicetools neopeptides per cell line and remove duplicate peptides
```{r}
#SKMEL5
SNAF_SKMEL5_neopeptide_seq <- SNAF_SKMEL5_neopeptide %>% select(Peptide)
splicetools_neopeptides_SKMEL5_seq <- splicetools_neopeptides_SKMEL5 %>% select(Peptide)
neopeptides_SKMEL5 <- rbind(SNAF_SKMEL5_neopeptide_seq, splicetools_neopeptides_SKMEL5_seq)
neopeptides_SKMEL5 <- distinct(neopeptides_SKMEL5)

#SKMEL24
SNAF_SKMEL24_neopeptide_seq <- SNAF_SKMEL24_neopeptide %>% select(Peptide)
splicetools_neopeptides_SKMEL24_seq <- splicetools_neopeptides_SKMEL24 %>% select(Peptide)
neopeptides_SKMEL24 <- rbind(SNAF_SKMEL24_neopeptide_seq, splicetools_neopeptides_SKMEL24_seq)
neopeptides_SKMEL24 <- distinct(neopeptides_SKMEL24)

#SKMEL28
SNAF_SKMEL28_neopeptide_seq <- SNAF_SKMEL28_neopeptide %>% select(Peptide)
splicetools_neopeptides_SKMEL28_seq <- splicetools_neopeptides_SKMEL28 %>% select(Peptide)
neopeptides_SKMEL28 <- rbind(SNAF_SKMEL28_neopeptide_seq, splicetools_neopeptides_SKMEL28_seq)
neopeptides_SKMEL28 <- distinct(neopeptides_SKMEL28)

#451Lu
SNAF_451Lu_neopeptide_seq <- SNAF_451Lu_neopeptide %>% select(Peptide)
splicetools_neopeptides_451Lu_seq <- splicetools_neopeptides_451Lu %>% select(Peptide)
neopeptides_451Lu <- rbind(SNAF_451Lu_neopeptide_seq, splicetools_neopeptides_451Lu_seq)
neopeptides_451Lu <- distinct(neopeptides_451Lu)

#1205Lu
SNAF_1205Lu_neopeptide_seq <- SNAF_1205Lu_neopeptide %>% select(Peptide)
splicetools_neopeptides_1205Lu_seq <- splicetools_neopeptides_1205Lu %>% select(Peptide)
neopeptides_1205Lu <- rbind(SNAF_1205Lu_neopeptide_seq, splicetools_neopeptides_1205Lu_seq)
neopeptides_1205Lu <- distinct(neopeptides_1205Lu)

#WM793
SNAF_WM793_neopeptide_seq <- SNAF_WM793_neopeptide %>% select(Peptide)
splicetools_neopeptides_WM793_seq <- splicetools_neopeptides_WM793 %>% select(Peptide)
neopeptides_WM793 <- rbind(SNAF_WM793_neopeptide_seq, splicetools_neopeptides_WM793_seq)
neopeptides_WM793 <- distinct(neopeptides_WM793)
```

```{r}
write.csv(neopeptides_SKMEL5, file = "neopeptides_SKMEL5.csv")
write.csv(neopeptides_SKMEL24, file = "neopeptides_SKMEL24.csv")
write.csv(neopeptides_SKMEL28, file = "neopeptides_SKMEL28.csv")
write.csv(neopeptides_451Lu, file = "neopeptides_451Lu.csv")
write.csv(neopeptides_1205Lu, file = "neopeptides_1205Lu.csv")
write.csv(neopeptides_WM793, file = "neopeptides_WM793.csv")
```


#Make UpSet plot
```{r}
devtools::install_github("krassowski/complex-upset")
```
```{r}
library(ComplexUpset)
library(dplyr)
```


```{r}
neopeptides_WM793 <- read.csv("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Peptides/neopeptides_WM793.csv")
neopeptides_SKMEL24 <- read.csv("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Peptides/neopeptides_SKMEL24.csv")
neopeptides_SKMEL28 <- read.csv("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Peptides/neopeptides_SKMEL28.csv")
neopeptides_SKMEL5 <- read.csv("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Peptides/neopeptides_SKMEL5.csv")
neopeptides_1205Lu <- read.csv("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Peptides/neopeptides_1205Lu.csv")
neopeptides_451Lu <- read.csv("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Peptides/neopeptides_451Lu.csv")
```



```{r}
library(ComplexHeatmap)
neopeptides_WM793 <- neopeptides_WM793$WM793
neopeptides_451Lu <- neopeptides_451Lu$X451Lu
neopeptides_1205Lu <- neopeptides_1205Lu$X1205Lu
neopeptides_SKMEL5 <- neopeptides_SKMEL5$SKMEL5
neopeptides_SKMEL24 <- neopeptides_SKMEL24$SKMEL24
neopeptides_SKMEL28 <- neopeptides_SKMEL28$SKMEL28

by_event = list(
  "1205Lu" =neopeptides_1205Lu, 
  "451Lu" =neopeptides_451Lu,
  WM793 =neopeptides_WM793,
  SKMEL5 =neopeptides_SKMEL5,
  SKMEL24 =neopeptides_SKMEL24,
  SKMEL28 =neopeptides_SKMEL28
)
by_event_matrix<- list_to_matrix(by_event)
by_event_matrix <-as.data.frame(by_event_matrix)


by_event_matrix <- by_event_matrix %>% dplyr::select("SKMEL5", "451Lu", "SKMEL28", "1205Lu", "WM793", "SKMEL24")
cell_line <- colnames(by_event_matrix)
```

```{r}

# Create the UpSet plot
upset_plot_events <- ComplexUpset::upset(by_event_matrix, cell_line, width_ratio=0.4, sort_sets=FALSE, sort_intersections=FALSE, max_size = 100000, name = NULL,intersections=list(
        c('451Lu', 'SKMEL28', 'SKMEL5'),
        c('1205Lu', 'WM793', 'SKMEL24'),
        c('1205Lu', 'WM793', 'SKMEL24', '451Lu', 'SKMEL28', 'SKMEL5')), 
        queries=list(
           upset_query(set ='451Lu', fill='red'),
            upset_query(set='SKMEL28', fill='red'),
             upset_query(set='SKMEL5', fill='red'),
           upset_query(set ='1205Lu', fill='blue'),
           upset_query(set ='WM793', fill='blue'),
            upset_query(set='SKMEL24', fill='blue')),
        set_sizes=(upset_set_size())
)

# Display the UpSet plot
print(upset_plot_events)
```


```{r}
pdf(file="upset_plot_neopeptide.pdf", width = 6, height = 6)
upset_plot_events
dev.off()
```



#Jaccard Similarity Comparison

```{r}
# Function to calculate Jaccard similarity index between two sets
jaccard_similarity <- function(set1, set2) {
  intersection <- sum(set1 & set2)  # Count elements common to both sets
  union <- sum(set1 | set2)          # Count unique elements in both sets
  return(intersection / union)       # Calculate Jaccard similarity index
}

# Create a matrix to store Jaccard similarity index between sets
n_sets <- ncol(by_event_matrix)
jaccard_matrix <- matrix(0, n_sets, n_sets)

# Calculate Jaccard similarity index between all pairs of sets
for (i in 1:n_sets) {
  for (j in 1:n_sets) {
    if (i == j) {
      jaccard_matrix[i, j] <- 1  # Set similarity to 1 when comparing the same set with itself
    } else if (i < j) {
      jaccard_matrix[i, j] <- jaccard_similarity(by_event_matrix[, i], by_event_matrix[, j])
      jaccard_matrix[j, i] <- jaccard_matrix[i, j]  # Since Jaccard index is symmetric
    }
  }
}

# Add column names to the Jaccard similarity matrix
colnames(jaccard_matrix) <- colnames(by_event_matrix)

# Print the Jaccard similarity matrix
print(jaccard_matrix)



```


```{r}
# Create the heatmap
library(pheatmap)
jaccard <- pheatmap(jaccard_matrix, 
          name = c("Jaccard Index"),
          col = colorRampPalette(c("white","orange", "red"))(100), 
          show_colnames = TRUE,
          show_rownames = TRUE,
          display_numbers = TRUE,
          angle_col = c("45"),
          treeheight_row = 0,
          treeheight_col = 0,
          fontsize_row = 8,
          fontsize_col = 8,
          legend = TRUE)
jaccard
```

```{r}
pdf(file="jaccard.pdf", width = 3.75, height = 3.5)
jaccard
dev.off()
```


#binding data

```{r}
HLA_athena <- read.csv("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Peptides/HLAathena.csv")

#HLA_athena <- HLA_athena %>%
  #filter(!Peptides %in% missing$Peptides)

NET_MHC <- read.csv("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Peptides/NETMHC.csv")

#NET_MHC <- NET_MHC %>%
  #filter(!Peptides %in% missing$Peptides)
```


```{r}
HLA_A11 <- HLA_athena %>% filter(prank.MSi_A1101 < 2) %>% distinct(Peptides, .keep_all = TRUE) 
HLA_B40 <- HLA_athena %>% filter(prank.MSi_B4001 < 2) %>% distinct(Peptides, .keep_all = TRUE)
HLA_C03 <- HLA_athena %>% filter(prank.MSi_C0304 < 2) %>% distinct(Peptides, .keep_all = TRUE)
  
NET_EL_A11 <- NET_MHC %>% filter(EL_Rank_A11 < 2) %>% distinct(Peptides, .keep_all = TRUE)
NET_EL_B40 <- NET_MHC %>% filter(EL_Rank_B40 < 2) %>% distinct(Peptides, .keep_all = TRUE)
NET_EL_C03 <- NET_MHC %>% filter(EL_Rank_C03 < 2) %>% distinct(Peptides, .keep_all = TRUE)


#A11
A11 <- inner_join(HLA_A11, NET_EL_A11, by = "Peptides")

A11$Label <- "A1101"

A11 <- unique(A11)

#B40
B40 <- inner_join(HLA_B40, NET_EL_B40, by = "Peptides")

B40$Label <- "B4001"

B40 <- unique(B40)

#C03
C03 <- inner_join(HLA_C03, NET_EL_C03, by = "Peptides")

C03$Label <- "C0304"

C03 <- unique(C03)


Filtered_neopeptides <- rbind(A11,B40,C03)

Filtered_neopeptides <- Filtered_neopeptides %>% rename(A1101_HLA = prank.MSi_A1101, B4001_HLA = prank.MSi_B4001, C0304_HLA = prank.MSi_C0304, A1101_EL_Rank = EL_Rank_A11, B4001_EL_Rank = EL_Rank_B40, C0304_EL_Rank = EL_Rank_C03)
```


#make bar plot
```{r}
table_names <- c("HLA_A11", "HLA_B40", "HLA_C03", "NET_EL_A11", "NET_EL_B40", "NET_EL_C03")
row_counts <- numeric(length(table_names))

# Loop through the tables and count rows
for (i in 1:length(table_names)) {
  row_counts[i] <- nrow(get(table_names[i]))
}

library(ggplot2)

# Create a data frame with table names, row counts, and groups
count_data <- data.frame(Table = table_names, Count = row_counts)
count_data$Group <- ifelse(grepl("HLA", count_data$Table), "HLA", 
                          ifelse(grepl("EL", count_data$Table), "EL", "BA"))
count_data$HLA <- ifelse(grepl("A11", count_data$Table), "A11", 
                          ifelse(grepl("B40", count_data$Table), "B40", "C03"))

# Create a grouped bar plot
count_hla <- ggplot(count_data, aes(x = Table, y = Count, fill = Group)) +
  geom_bar(stat = "identity", position = "dodge", color = "black") +
  scale_fill_manual(values = c("cornflowerblue", "gray95")) +
  facet_wrap( ~ HLA , scales = "free_x")+
  labs(title = "Predicted Strong Binders", x = NULL, y = "# Predicted Strong Binders") +
  theme_bw() +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 700))
  theme(axis.text.x = element_blank())  # Rotate x-axis labels if needed
  
  
count_hla <- count_hla + theme(axis.text.x = element_blank(), axis.ticks.x=element_blank(), plot.title = element_text(hjust = 0.5))
count_hla
```

```{r}
pdf(file="neopeptide_count.pdf", width = 6, height = 5)
count_hla
dev.off()
```

#make venn diagram
```{r}
install.packages("eulerr")
```

```{r}
library(eulerr)
library(VennDiagram)
```

```{r}
HLA_A11_peptide <- HLA_A11$Peptide
NET_EL_A11_peptide <- NET_EL_A11$Peptide

# Create a list of sets
sets_list <- list(
  HLA_A11_peptides = HLA_A11_peptide,
  NET_EL_A11_peptides = NET_EL_A11_peptide
)

venn.diagram(
  x = sets_list,
  category.names = c("HLA" , "EL"),
  filename = 'A11_venn.png',
  output=TRUE
)
```


```{r}
A11_venn <- euler(c(HLAthena = 280, NET_EL = 259, 
                "HLAthena&NET_EL" = 210))
A11_venndiagram <- plot(A11_venn, quantities = TRUE, fill = c("gray95", "cornflowerblue"))
A11_venndiagram
```
```{r}
pdf(file="A11_venndiagram.pdf", width = 4, height = 4)
A11_venndiagram
dev.off()
```

```{r}
HLA_B40_peptide <- HLA_B40$Peptide
NET_EL_B40_peptide <- NET_EL_B40$Peptide

# Create a list of sets
sets_list <- list(
  HLA_B40_peptides = HLA_B40_peptide,
  NET_EL_B40_peptides = NET_EL_B40_peptide
)

venn.diagram(
  x = sets_list,
  category.names = c("HLA" , "EL"),
  filename = 'B40_venn.png',
  output=TRUE
)
```


```{r}
B40_venn <- euler(c(HLAthena = 466, NET_EL = 237, 
                "HLAthena&NET_EL" = 192))
B40_venndiagram <- plot(B40_venn, quantities = TRUE, fill = c("gray95", "cornflowerblue"))
B40_venndiagram
```
```{r}
pdf(file="B40_venndiagram.pdf", width = 4, height = 4)
B40_venndiagram
dev.off()
```

```{r}
HLA_C03_peptide <- HLA_C03$Peptide
NET_EL_C03_peptide <- NET_EL_C03$Peptide

# Create a list of sets
sets_list <- list(
  HLA_C03_peptides = HLA_C03_peptide,
  NET_EL_C03_peptides = NET_EL_C03_peptide
)

venn.diagram(
  x = sets_list,
  category.names = c("HLA" , "EL"),
  filename = 'C03_venn.png',
  output=TRUE
)
```


```{r}
C03_venn <- euler(c(HLAthena = 230, NET_EL =345, 
                "HLAthena&NET_EL" = 156))
C03_venndiagram <- plot(C03_venn, quantities = TRUE, fill = c("gray95", "cornflowerblue"))
C03_venndiagram
```
```{r}
pdf(file="C03_venndiagram.pdf", width = 4, height = 4)
C03_venndiagram
dev.off()
```


#mass spec data
```{r}
# Create a data frame with the provided data
data_rep1 <- data.frame(
  replicate = factor(rep(c("DMSO R1", "Corin R1"), each = 4), levels = c("DMSO R1", "Corin R1")),
  time = rep(8:11, times = 2),
  count = c(1848, 7972, 3185, 2800,   # DMSO R1
            2706, 9512, 3581, 3202) # Corin R1
)


data_rep2 <- data.frame(
  replicate = factor(rep(c("DMSO R2", "Corin R2"), each = 4), levels = c("DMSO R2", "Corin R2")),
  time = rep(8:11, times = 2),
  count = c(
            2516, 9450, 3830, 3248,  # DMSO R2
            
            2916, 10790, 4311, 3513) # Corin R2
)

# Combine the data for replicate 1 and replicate 2
combined_data <- rbind(data_rep1, data_rep2)

# Create a new column to distinguish replicates
combined_data$replicate_group <- ifelse(combined_data$replicate %in% c("DMSO R1", "Corin R1"), "Replicate 1", "Replicate 2")

# Create histogram plot faceted by replicate group
MSFragger_faceted <- ggplot(combined_data, aes(x = time, y = count, fill = replicate)) +
  geom_bar(stat = "identity", position = "dodge") +
  labs(title = "MSFragger - Replicates Comparison",
       x = "Peptide Length", y = "Count") +
  theme_bw() +
  scale_y_continuous(limits = c(0, 11000), expand = expansion(add = c(0, 0))) +  # Set y-axis limit to 11,000 and remove the gap at 0
  scale_x_continuous(breaks = unique(combined_data$time)) +  # Show all x-axis tick mark values
  facet_wrap(~ replicate_group, nrow = 1)  # Facet plots by replicate group, in a single row

# Print the faceted plot
MSFragger_faceted
```
```{r}
ggsave("MSFragger.pdf", MSFragger_faceted, width = 4, height = 3)

```




```{r}
library(Biostrings)
library(stringr)
```

#combine splicetools data back to neopeptides
```{r}
# Read FASTA file and extract sequences and identifiers
fasta_file <- "/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/SpliceTools/Splicetools Outputs/Neopeptides/SE_NEOPEPTIDES_SKMEL5_Filtered7.fasta"

fasta_seqs <- readLines(fasta_file)

# Extract identifiers and sequences from FASTA file
fasta_ids <- fasta_seqs[str_starts(fasta_seqs, ">")]
fasta_seqs <- fasta_seqs[!str_starts(fasta_seqs, ">")]

# Create a new column in Filtered_neopeptides to store associated identifiers
Filtered_neopeptides$Associated_Ids <- NA

# Iterate over each peptide in the list
for (peptide in Filtered_neopeptides$Peptides) {
  # Initialize vector to store matching identifiers
  matching_ids <- character(0)
  
  # Iterate over each sequence in the FASTA file
  for (i in seq_along(fasta_seqs)) {
    if (str_detect(fasta_seqs[i], peptide)) {
      matching_ids <- c(matching_ids, fasta_ids[i])
    }
  }
  
  # If matches were found, concatenate identifiers into a single string
  if (length(matching_ids) > 0) {
    Filtered_neopeptides$Associated_Ids[Filtered_neopeptides$Peptides == peptide] <- paste(matching_ids, collapse = ", ")
  }
}

# Print the modified Filtered_neopeptides data frame
print(Filtered_neopeptides)
```

```{r}
# Separate the string in Associated_Ids by the first and second comma
split_ids <- strsplit(Filtered_neopeptides$Associated_Ids, ",", fixed = TRUE)

# Extract the number between the first and second comma
Filtered_neopeptides$upstreamEE <- sapply(split_ids, function(x) {
  if(length(x) > 1) {
    as.numeric(x[2])
  } else {
    NA
  }
})

# Extract the number between the first and second comma
Filtered_neopeptides$downstreamES <- sapply(split_ids, function(x) {
  if(length(x) > 1) {
    as.numeric(x[3])
  } else {
    NA
  }
})
```


```{r}
SKMEL5_SE_rmats <- read.csv("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/rmats/processed_rmats/filtered/SE_SKMEL5_filtered.csv")

Filtered_neopeptides_SKMEL5 <- merge(Filtered_neopeptides, SKMEL5_SE_rmats, by.x = c("upstreamEE", "downstreamES"), by.y = c("upstreamEE", "downstreamES"), all.x = FALSE)

Filtered_neopeptides_SKMEL5 <- Filtered_neopeptides_SKMEL5[, !(names(Filtered_neopeptides_SKMEL5) %in% c("X", "ID", "ID.1"))]

unique_Filtered_neopeptides_SKMEL5 <- unique(Filtered_neopeptides_SKMEL5)

```

```{r}
unique_Filtered_neopeptides_SKMEL5 <- unique_Filtered_neopeptides_SKMEL5 %>%
  rowwise() %>%
  mutate(IJC_avg = mean(as.numeric(strsplit(IJC_SAMPLE_2, ",")[[1]])),
         SJC_avg = mean(as.numeric(strsplit(SJC_SAMPLE_2, ",")[[1]])),
         junctions = IJC_avg + SJC_avg)
```

```{r}
unique_Filtered_neopeptides_SKMEL5 <- unique_Filtered_neopeptides_SKMEL5 %>% dplyr::select(Peptides, geneSymbol, A1101_HLA, A1101_EL_Rank, B4001_HLA, B4001_EL_Rank, C0304_HLA, C0304_EL_Rank, junctions, IncLevelDifference) %>% rename(dPSI = IncLevelDifference)
```

```{r}
unique_Filtered_neopeptides_SKMEL5 <- unique_Filtered_neopeptides_SKMEL5 %>%
  mutate(Annotation = "Splicetools")
```


#adding in SNAF data
#first let's combine the junction, peptide, and gene information merging on the gene information

```{r}
SNAF_junction <- read.delim("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/SNAF/junction_counts.txt")
```

```{r}
# Calculate the average of columns J1.bed and J2.bed along rows
averages <- rowMeans(SNAF_junction[, c("J1.bed", "J2.bed")], na.rm = TRUE)

# Create a new column called junctions with the calculated averages
SNAF_junction$junctions <- averages
```

```{r}
SNAF_junction <- SNAF_junction %>% rename(Region = X) %>% dplyr::select(Region, junctions)
```

```{r}
SNAF_peptides <- read.delim("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/SNAF/peptides_all.txt")

SNAF_peptides <- tidyr::separate(SNAF_peptides, X, into = c("Peptides", "Region"), sep = ",")

SNAF_peptides <- SNAF_peptides %>% dplyr::select(Peptides, Region, symbol, tumor_specificity_mean, samples)
```

```{r}
SNAF_junction_peptides <- inner_join(SNAF_peptides, SNAF_junction, by = "Region")
```

#next combine the resulting table with dPSI values merging on gene information

```{r}
SNAF_PSI <- read.csv("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/SNAF/SNAF_PSI.csv")

SNAF_PSI <- separate(SNAF_PSI, UID, into = c("symbol", "Full_Region"), sep = ":", extra = "merge", remove = FALSE)

SNAF_PSI <- separate(SNAF_PSI, Full_Region, into = c("Region", "Extra"), sep = "\\|", remove = FALSE)

SNAF_PSI <- SNAF_PSI %>% dplyr::select(Region, dPSI, EventAnnotation)
```

```{r}
SNAF_data <- inner_join(SNAF_junction_peptides, SNAF_PSI, by = "Region")
```

```{r}
write.csv(SNAF_data, file = "SNAF_PSI.csv")
```

#combine with Filtered_neopeptides
```{r}
Filtered_neopeptides_SNAF <- inner_join(Filtered_neopeptides, SNAF_data, by = "Peptides")
Filtered_neopeptides_SNAF <- Filtered_neopeptides_SNAF %>% filter(abs(dPSI) > 0.1) %>% dplyr::select(Peptides, symbol, A1101_HLA, A1101_EL_Rank, B4001_HLA, B4001_EL_Rank, C0304_HLA, C0304_EL_Rank,junctions, dPSI) %>% rename(geneSymbol = symbol)

Filtered_neopeptides_SNAF <- Filtered_neopeptides_SNAF %>%
  mutate(Annotation = "SNAF")
```

```{r}
Neopeptides_heatmap <- rbind(Filtered_neopeptides_SNAF,unique_Filtered_neopeptides_SKMEL5) 
```

```{r}
Neopeptides_heatmap$dPSI <- abs(Neopeptides_heatmap$dPSI)
```

```{r}
Neopeptides_heatmap <- Neopeptides_heatmap %>%
  mutate_at(vars(3:8), ~ -log10(.))
```


```{r}
Neopeptides_heatmap$Best_Rank <- apply(Neopeptides_heatmap[, 3:8], 1, max)
```

```{r}
# Extract column names without the first two columns (ID and Peptides)
column_names <- names(Neopeptides_heatmap)[-c(1, 2)]

# Find the index of the column with the maximum value in each row
best_rank_column_index <- max.col(Neopeptides_heatmap[, 3:8], "first")

# Report the column name with the best rank for each row
Neopeptides_heatmap$Best_HLA <- column_names[best_rank_column_index]

Neopeptides_heatmap$Best_HLA <- sub("_.*", "", Neopeptides_heatmap$Best_HLA)

```


```{r}
Neopeptides_heatmap <- Neopeptides_heatmap %>%
  mutate(Score = Best_Rank * junctions * dPSI)
```


#find the top 20 peptides based on score
```{r}
# Arrange the data frame based on the Score column in descending order
Neopeptides_heatmap_sorted <- Neopeptides_heatmap %>%
  arrange(desc(Score))

# Add a column with ascending numbers
Neopeptides_heatmap_sorted$ascending_numbers <- seq_len(nrow(Neopeptides_heatmap_sorted))

# Combine the ascending numbers column with the Peptides column
Neopeptides_heatmap_sorted$Ranked_Peptides <- paste0(Neopeptides_heatmap_sorted$ascending_numbers, "_", Neopeptides_heatmap_sorted$Peptides)

top_20_peptides <- head(Neopeptides_heatmap_sorted$Ranked_Peptides, 21)
print(top_20_peptides)

```

#heatmap of binding
```{r}
heatmap_data <- Neopeptides_heatmap_sorted %>% dplyr::select(Peptides, A1101_HLA, A1101_EL_Rank, B4001_HLA, B4001_EL_Rank, C0304_HLA, C0304_EL_Rank, Annotation)
```

```{r}
library(circlize)
library(ComplexHeatmap)


# Create a heatmap object with the custom color palette
heatmap_rank <- Heatmap(
  heatmap_data[,2:7],
  name = "-Log10(Rank)",
  cluster_rows = T,
  cluster_columns = FALSE,
  show_row_names = FALSE,
  show_column_names = TRUE,
  #right_annotation = ha,
  show_row_dend = FALSE,
  column_names_rot = 45,
  column_names_side = "top",
  #rect_gp = gpar(col = "black", lwd = 0.01),
  border_gp = gpar(col = "black"),
  row_names_gp = gpar(fontsize = 3),
  left_annotation = rowAnnotation(Tool = Neopeptides_heatmap$Annotation),
  heatmap_legend_param = list(
                                      legend_direction = "horizontal", 
                                      legend_width = unit(3, "cm")
   ),
  col = colorRamp2(c(-2, 4), colors = c("white","dodgerblue3"))
)


# Draw the heatmap
draw(heatmap_rank,  heatmap_legend_side="bottom")

```
```{r}
pdf(file="heatmaps_peptide.pdf", width = 5, height = 9)
heatmap_rank
dev.off()
```


```{r}
PSI <- Neopeptides_heatmap_sorted %>% dplyr::select(Ranked_Peptides, dPSI)
PSI <- PSI[,-1]
```

```{r}
# Create a heatmap object with the custom color palette
heatmap_PSI <- Heatmap(
  PSI,
  name = "|deltaPSI|",
  cluster_rows = TRUE,
  cluster_columns = FALSE,
  show_row_names = FALSE,
  show_column_names = TRUE,
  row_names_gp = gpar(fontsize = 3),
  column_names_rot = 45,
  column_names_side = "top",
  #rect_gp = gpar(col = "black", lwd = 0.01),
  border_gp = gpar(col = "black"),
  heatmap_legend_param = list(
                                      legend_direction = "horizontal", 
                                      legend_width = unit(3, "cm"),
                                      at = seq(0, 1, by = 0.25)
                                     ),
  col = colorRamp2(c(0.1, 1), colors = c("white", "dodgerblue3"))

)

# Draw the heatmap
draw(heatmap_PSI, heatmap_legend_side="bottom")

```




```{r}
Count <- Neopeptides_heatmap_sorted %>% dplyr::select(Ranked_Peptides, junctions)
```

```{r}
# Create a heatmap object with the custom color palette
library(ComplexHeatmap)
library(circlize)

# Define the specific_row_names
specific_row_names <- c("1_RLLPPIFGK","2_RLLPPIFGKK","3_RLLPPIFGK","4_FSINLQTSL","5_VELEDHVML","6_RLLPPIFGKK","7_FSINLQTSL","8_REDPVPPQL","9_LTCTEILLK","10_YALRTSTL","11_SIYVPCVAK","12_KVELEDHVML","13_FSINLQTSLL","14_YALRTSTL","15_SIYVPCVAK","16_ATSVPSCRK","17_SERLPLTYL","18_AEHAHRVPL","19_LTCTEILLK","20_REQGGTAGL","21_WAHPTKGLGL","327_RSQGWLFLR", "123_RTVRPMGLVK")

# Find row indices for specific_row_names and keep the order
row_indices <- match(specific_row_names, Count$Ranked_Peptides)
specific_row_names

ha <- rowAnnotation(
  foo = anno_mark(
    at = row_indices,
    labels = specific_row_names,
    labels_gp = gpar(col = "black", fontsize = 9)  # Adjust the fontsize here
  )
)

# Create the heatmap
heatmap_Count <- Heatmap(
  Count[,-1],
  name = "IJC + SJC",
  cluster_rows = TRUE,
  cluster_columns = FALSE,
  show_row_names = FALSE,
  show_column_names = TRUE,
  column_names_rot = 45,
  right_annotation = ha,
  column_names_side = "top",
  border_gp = gpar(col = "black"),
  row_names_gp = gpar(fontsize = 4),
  heatmap_legend_param = list(
                                      legend_direction = "horizontal", 
                                      legend_width = unit(3, "cm")
                                     ),
  col = colorRamp2(c(0, 1150), colors = c("white", "dodgerblue3")),
)

# Draw the heatmap
draw(heatmap_Count, heatmap_legend_side="bottom")
```
```{r}
heatmap_neopeptide <- heatmap_rank+heatmap_PSI+heatmap_Count
draw(heatmap_neopeptide,  heatmap_legend_side="bottom")
```

```{r}
pdf(file="heatmaps_peptide_try.pdf", width = 6, height = 10)
draw(heatmap_neopeptide,heatmap_legend_side="bottom")
dev.off()
```


#venn diagram for corin-exclusive peptides identified by mass spec and predicted peptides

```{r}
corin_exc <- read.csv("/Users/rjfisher/Library/Mobile Documents/com~apple~CloudDocs/BUSM/Alani Lab/DATA/CoREST Splicing/Peptides/corin_exc_peptides.csv")
```

```{r}
Neopeptides_heatmap_sorted_unique <- Neopeptides_heatmap_sorted %>% distinct(Peptides)
```

```{r}
predicted_peptides <- Neopeptides_heatmap_sorted$Peptide
corin_exc_peptides <- corin_exc$Peptide

# Create a list of sets
sets_list <- list(
  predicted_peptides = predicted_peptides,
  corin_exc_peptides = corin_exc_peptides
)

venn.diagram(
  x = sets_list,
  category.names = c("predicted" , "mass spec"),
  filename = 'Predicted_massspec.png',
  output=TRUE
)
```


```{r}
Predicted_massspec <- euler(c(Mass_Spec = 7460, Predicted = 554, 
                "Mass_Spec&Predicted" = 2))
Predicted_massspec_venn <- plot(Predicted_massspec, quantities = TRUE, fill = c("gray95", "cornflowerblue"))
Predicted_massspec_venn
```


```{r}
pdf(file="Predicted_massspec_venn.pdf", width = 4, height = 4)
Predicted_massspec_venn
dev.off()
```

```{r}
missing <- anti_join(Filtered_neopeptides, Neopeptides_heatmap_sorted_unique, by = "Peptides")
```

```{r}
write.csv(Neopeptides_heatmap_sorted, file = "Neopeptides_all.csv")
write.csv(Neopeptides_heatmap_sorted_unique, file = "Neoppetides_all_unique.csv")
```

